### PR TITLE
RD-3110 Enhance Deploy On modal to allow providing a suffix for new deployments

### PR DESCRIPTION
**Pull request by qooban**
## Description

Within this PR new field is added to "Deploy On" modal. This is text input field which is then used to create deployment name for newly created deployments.

### Deploy On modal

Screenshot from 2021-09-14 09-41-39

#### Deploy modal in Create Deployment Button widget
Screenshot from 2021-09-14 09-42-26

#### Deploy modal in Blueprints widget
Screenshot from 2021-09-14 09-42-57

## Checklist
- [x] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires a documentation update.
- [x] I added proper labels to this PR.

## Tests
[Stage-UI-System-Test/1252](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/1252/pipeline) (14.09.2021, 09:44:01 CEST) - PASSED

## Documentation
Deployments View widget description update PR: https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/1949 .
I'd like to merge it first and then update Deployments View widget's `README.md` within this PR.

### CODE REVIEW

File: widgets/common/src/GenericDeployModal.tsx [hunk 1]
```diff
@@ -353,6 +326,35 @@ class GenericDeployModal extends React.Component {
                             </Form.Field>
                         )}
 
+                        {showDeploymentNameInput && (
```
Comment (qooban) [inline widgets/common/src/GenericDeployModal.tsx:329]: I have changed the order of the fields. Now Deployment Name and Deployment ID are displayed after Blueprint dropdown (which is shown only in Create Deployment Button widget). I think that's OK. Let me know if you see it differently.

File: app/translations/en.json [hunk 2]
```diff
@@ -893,6 +893,12 @@
                         "title": "Deploy On",
                         "modal": {
                             "header": "Deploy on",
+                            "inputs": {
+                                "name": {
+                                    "label": "Name",
```
Comment (kubama) [inline app/translations/en.json:898]: Wouldn't 'Name suffix' be a more accurate label?

To be honest, I find the whole modal quite confusing, and the 'Name' field even adds to the confusion. Without reading the docs I think it's really hard to guess what the "Deploy On" term actually means in its context. The new 'Name' field makes this modal similar to regular deployment modal making it even harder to understand its actual purpose. Not necessarily as part of this ticket, but I would consider the following:
1. Changing modal title to be at least a little bit more descriptive, there is no reason to keep it as short as a button label.
2. Providing a text box that after filling in the Name would shorty describe what deployments will be created, something like a short message saying "The following deployments will be created: env1-name, env2-name, env3-name".

**Comment by qooban** [comment]
@kubama Documentation for Deployments View widget updated (`README.md` file), so it's ready for review.

File: app/translations/en.json (hunk 2 above)
Comment (qooban) [reply app/translations/en.json:898]: 1. It's a pity you already approved documentation change, but I can change the field name to 'Name suffix', make sense to me. @alexmolev what do you think? 

2. Regarding general change of the modal - please either create a ticket or discuss with @alexmolev or @ilanyaniv . I have no strong opinion here as for now.

**Review by kubama — APPROVED** [review]

**Comment by qooban** [comment]
Consulted with Alex and changed 'Name' to 'Name suffix'.
Documentation PR should be merged first: https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/1950 .
System test for Deployments View widget triggered locally - passed. Not planning full run again.

**Comment by qooban** [comment]
Documentation for Deployments View widget updated (README.md file), so it's ready for review.

**Review by kubama — APPROVED** [review]

**Pull request merged**